ibis-server/app/routers/v2/connector.py

```diff
@@ -271,16 +271,17 @@ async def dry_plan_for_data_source(
 async def model_substitute(
     data_source: DataSource,
     dto: TranspileDTO,
+    request: Request,
```

Comment:
I think `headers` has been injected at L276. Why do we need to inject `request` again?